commit 8d396b1de125bd5c5f895c5b290c373d0a2f137a
Author: Amos Jeffries <yadij@users.noreply.github.com>
Date:   2019-11-19 01:06:56 +1300

    Fix detection of sys/sysctl.h detection (#511)
    
    Make sure we test the EUI specific headers using same flags
    chosen for final build operations. This should make the
    test detect the header as unavailable if the user options
    would make the compiler #warning be a fatal error later.

diff --git a/configure.ac b/configure.ac
index 2a6fa6a..6c93de0 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1131,6 +1131,10 @@ AC_ARG_ENABLE(eui,
 SQUID_YESNO([$enableval],[--disable-eui expects no arguments])
 ])
 if test "x${enable_eui:=yes}" = "xyes" ; then
+  SQUID_STATE_SAVE(LIBEUI)
+  # GLIBC 2.30 deprecates sysctl.h. Test with the same flags that (may) break includes later.
+  CFLAGS=$SQUID_CFLAGS
+  CXXFLAGS=$SQUID_CXXFLAGS
   case "$squid_host_os" in
     linux|solaris|freebsd|openbsd|netbsd|cygwin)
       ${TRUE}
@@ -1170,6 +1174,7 @@ include <windows.h>
 #include <sys/param.h>
 #endif
   ]])
+  SQUID_STATE_ROLLBACK(LIBEUI)
 fi
 AC_SUBST(EUILIB)
 AC_MSG_NOTICE([EUI (MAC address) controls enabled: $enable_eui])
